### PR TITLE
SC WS: dry run a contract

### PR DESCRIPTION
**Pull request by velzevur**
PT [SC WS: Dry run a contract call ](https://www.pivotaltracker.com/story/show/159406104)
Those are the changes introduced in #2091 but opened against `minerva`

### CODE REVIEW

File: docs/release-notes/next/PT-159406104_SC_WS_dry_call_contract.md [hunk 1]
```diff
@@ -0,0 +1 @@
+* Adds anew WebSocket method for dry-running off-chain contracts: `channels.dry_run.call_contract`.
```
Comment (lucafavatella) [inline docs/release-notes/next/PT-159406104_SC_WS_dry_call_contract.md]:
```suggestion
* Adds a new WebSocket method for dry-running off-chain contracts: `channels.dry_run.call_contract`.
```

**Comment by velzevur** [comment]
Rebased on `master`

File: apps/aehttp/test/aehttp_integration_SUITE.erl [hunk 2]
```diff
@@ -4146,13 +4146,18 @@ sc_ws_oracle_contract_(Owner, GetVolley, ConnPid1, ConnPid2,
...
             {UpdateVolley, UpdaterConnPid, _UpdaterPubKey} = GetVolley(Who),
+            {ok, #{<<"value">> := R0}} =
+                dry_call_a_contract(Fun, Args, ContractPubKey,
+                                    EncodedCode, UpdaterConnPid,
+                                    Config, ReturnType),
             Tx = call_a_contract(Fun, Args,
                                  ContractPubKey, EncodedCode,
                                  UpdaterConnPid, UpdateVolley, Config),
-            #{<<"value">> := R} =
+            #{<<"value">> := R} = % same value as the
```
Comment (lucafavatella) [inline apps/aehttp/test/aehttp_integration_SUITE.erl]: Incomplete comment?

File: apps/aehttp/src/sc_ws_api_jsonrpc.erl [hunk 3]
```diff
@@ -292,6 +292,28 @@ process_request(#{<<"method">> := <<"channels.get.contract_call">>,
...
+                                             amount       => Amount,
+                                             call_data    => CallData}) of
+                {ok, Call} ->
+                  {reply, #{ action     => <<"dry_run">>
+                           , tag        => <<"call_contract">>
+                           , {int,type} => reply
+                           , payload    => aect_call:serialize_for_client(Call) }};
+                {error, _Reason} = Err -> Err
+            end;
+        _ -> {error, broken_code}
```
Comment (lucafavatella) [inline apps/aehttp/src/sc_ws_api_jsonrpc.erl]: ... or call data? Ok though.

File: apps/aehttp/src/sc_ws_api_jsonrpc.erl [hunk 4]
```diff
@@ -292,6 +292,28 @@ process_request(#{<<"method">> := <<"channels.get.contract_call">>,
...
+            case aesc_fsm:dry_run_contract(FsmPid,
+                                           #{contract     => Contract,
+                                             abi_version  => ABIVersion,
+                                             amount       => Amount,
+                                             call_data    => CallData}) of
+                {ok, Call} ->
+                  {reply, #{ action     => <<"dry_run">>
+                           , tag        => <<"call_contract">>
+                           , {int,type} => reply
+                           , payload    => aect_call:serialize_for_client(Call) }};
```
Comment (lucafavatella) [inline apps/aehttp/src/sc_ws_api_jsonrpc.erl]: May interact with https://github.com/aeternity/aeternity/pull/2201/files?utf8=%E2%9C%93&diff=split&w=1#diff-89b8ec7c2f28f3e3c7d21ca9bf070bb7R3037 though I see no evident conflict.

Comment (velzevur) [reply apps/aehttp/src/sc_ws_api_jsonrpc.erl]: Rebased on top of `master` to check if there is a hidden dependency

**Pull request merged**